master/buildbot/reporters/bitbucketserver.py

```diff
@@ -125,6 +129,203 @@ def send(self, build):
+    name = "BitbucketServerCoreAPIStatusPush"
+    secrets = ["token", "auth"]
+
+    def checkConfig(self, base_url, token=None, auth=None, **kwargs):
+        if not base_url:
+            config.error("Parameter base_url has to be given")
+        if token is not None and auth is not None:
+            config.error("Only one authentication method can be given "
+                         "(token or auth)")
+        super().checkConfig(**kwargs)
+
+    @defer.inlineCallbacks
+    def reconfigService(self, base_url, token=None, auth=None,
+                        statusName=None, statusSuffix=None, startDescription=None,
+                        endDescription=None, key=None, parentName=None,
+                        buildNumber=None, ref=None, duration=None,
+                        testResults=None, verbose=False, debug=None,
+                        verify=None, **kwargs):
+        yield super().reconfigService(wantProperties=True, **kwargs)
+        self.status_name = statusName
+        self.status_suffix = statusSuffix
+        self.start_description = startDescription or 'Build started.'
+        self.end_description = endDescription or 'Build done.'
+        self.key = key or Interpolate('%(prop:buildername)s')
+        self.parent_name = parentName
+        self.build_number = buildNumber or Interpolate('%(prop:buildnumber)s')
+        self.ref = ref
+        self.duration = duration
+
+        if testResults:
+            self.test_results = testResults
+        else:
+            @util.renderer
+            def r_testresults(props):
+                failed = props.getProperty("tests_failed", 0)
+                skipped = props.getProperty("tests_skipped", 0)
+                successful = props.getProperty("tests_successful", 0)
+                if any([failed, skipped, successful]):
+                    return {
+                        "failed": failed,
+                        "skipped": skipped,
+                        "successful": successful
+                    }
+                return None
+            self.test_results = r_testresults
+
+        self.verbose = verbose
+        headers = {}
+        if token:
+            headers["Authorization"] = "Bearer {}".format(token)
+        self._http = yield httpclientservice.HTTPClientService.getService(
+            self.master, base_url, auth=auth, headers=headers, debug=debug,
+            verify=verify)
+
+    def createStatus(self, proj_key, repo_slug, sha, state, url, key, parent,
+                     build_number, ref, description, name, duration,
+                     test_results):
+        payload = {
+            'state': state,
+            'url': url,
+            'key': key,
+            'parent': parent,
+            'ref': ref,
+            'buildNumber': build_number,
+            'description': description,
+            'name': name,
+            'duration': duration,
+            'testResults': test_results
+        }
+
+        if self.verbose:
```

Comment:
I'm wondering if this could be named `debug`? Other buildbot components use that name as convention. I see `verbose` used in bitbucketserver.py, this is perhaps how it came here :-)

Reply:
Turns out `debug` enables much more verbose HTTP logging. So no need to do anything.